### DISCUSSION TITLE
Calculate rumple index as a raster

### DISCUSSION THREAD

**Discussion by ncorroyez** [r-lidar/lidR#783] [Q&A]
Hello,

I've calculated a CHM of a LAS catalog. Now I want to extract the rumple index from this CHM. Or, when I do:

`rumple_index <- rumple_index(chm)`
I get a single numerical value: one value for the whole raster. I would like to get a full raster of same dimensions and resolution than my CHM. I looked at terra focal and app but didn't manage to make them work. Any ideas on that ?

Cheers

**Comment by Jean-Romain** [comment]
rumple index requires to aggregate many cells. You cannot get a rumple index with the same resolution than the CHM. Or maybe with a moving windows. You could take 20 x 20 cells on a moving windows and assign the value to the central pixel. then move the windows by one pixel. This is what `focal` does but `focal` does not not keep track of the spatial arrangement (XY coordinates) of the pixels, thus won't be able to process `rumple_index`. Thus you have to code it yourself.
